doc/DESIGN.md

```diff
@@ -1,47 +1,53 @@
-# DESIGN ##
+# DESIGN #
 
-This document describes the design of the c-api and gives
-hints for binding writers. It does not aim to plugin writers
-because this detail is hidden from the programmer and is
-Elektra specific.
+This document describes the design of the C-API and provides hints for
+binding writers. It is not aimed at plugin writers, since it does not
+talk about the implementation details of Elektra.
 
-Elektra follows the design principles to make it hard to use
-it wrong and fully aims towards an easy to use API for
-programmers reading and writing configuration. The data structures
-are optimized to get, set and lookup values easily and fast.
+Elektra follows two design principles:
 
-The idea is that the KDB API is not only implemented by Elektra.
-Elektra provides a full blown architecture to really support
-modern Linux Systems, but comes with some overhead. In this
-document the KDB API is described. But sometimes there are
-hints for elektra specific conventions.
+1. Make it hard to use the API the wrong way, and
+2. aim towards an easy to use API for programmers reading and writing
+   configuration.
 
+Elektra's data structures are optimized to get, set and lookup values
+easily and fast.
 
-## Data Structures ##
+The idea is, that the KDB API is not only implemented by Elektra.
+Elektra provides a full blown architecture to really support modern
+Linux Systems, but comes with some overhead. This document describes
+the `KDB` API. It also contains some hints about Elektra specific
+conventions.
 
-Key, KeySet and KDB datastructures are defined in kdbprivate.h
-to remain ABI compatible when something is added to a struct. That means
-it is not possible to put one of elektra's datastructures on the stack.
-You must use the memory management facilities mentioned in the next chapter.
+## Data Structures ##
 
+The `Key`, `KeySet` and `KDB` data structures are defined in
+`kdbprivate.h` to remain ABI compatible when something is added to a
```

Comment:
I change the line to:

> The `Key`, `KeySet` and `KDB` data structures are defined in
> `kdbprivate.h` to remain ABI compatible when one of them is changed.

Hope that works for you. If not, then please just comment below.